holidays/countries/cocos_islands.py

```diff
@@ -0,0 +1,210 @@
+        * <https://en.wikipedia.org/wiki/Public_holidays_in_the_Cocos_(Keeling)_Islands>
+        * <https://www.infrastructure.gov.au/territories-regions-cities/territories/indian-ocean-territories/community-bulletins>
+        * <https://www.infrastructure.gov.au/territories-regions-cities/territories/indian-ocean-territories/gazettes-bulletins>
+        * <https://publicholidays.asia/cocos-keeling-islands/>
+    """
+
+    country = "CC"
+    default_language = "coa_CC"
+    # %s (observed).
+    observed_label = tr("%s (disambut)")
+    supported_languages = ("coa_CC", "en_AU", "en_US")
+    start_year = 1955
+
+    def __init__(self, *args, **kwargs):
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        islamic_show_estimated = kwargs.pop("islamic_show_estimated", True)
+        IslamicHolidays.__init__(
+            self, cls=CocosIslandsIslamicHolidays, show_estimated=islamic_show_estimated
+        )
+        kwargs.setdefault("observed_rule", SAT_SUN_TO_NEXT_MON)
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # New Year's Day.
+        name = tr("Hari Tahun Baru")
+        self._add_observed(self._add_new_years_day(name))
+        self._add_observed(self._next_year_new_years_day, name=name)
+
+        # Australia Day.
+        self._add_observed(self._add_holiday_jan_26(tr("Hari Australia")))
+
+        # Islamic New Year.
+        if self._year < 2020:
+            self._add_islamic_new_year_day(tr("Tahun Baru Hijriah"))
+
+        # Eid al-Fitr.
+        self._add_eid_al_fitr_day(tr("Hari Raya Puasa"))
+
+        # Good Friday.
+        self._add_good_friday(tr("Jumat Agung"))
+
+        # Easter Monday.
+        self._add_easter_monday(tr("Isnin Paskah"))
+
+        # Self Determination Day.
+        if self._year >= 1984:
+            self._add_observed(self._add_holiday_apr_6(tr("Hari Penentuan Diri")))
+
+        # ANZAC Day.
+        self._add_observed(self._add_holiday_apr_25(tr("Hari ANZAC")))
+
+        # Eid al-Adha.
+        self._add_eid_al_adha_day(tr("Hari Raya Haji"))
+
+        # King's Birthday.
+        name = tr("Hari Ulang Tahun Raja")
+        self._add_holiday_2nd_mon_of_jun(name)
+
+        # Prophet Muhammad's Birthday.
+        self._add_mawlid_day(tr("Hari Maulaud Nabi"))
+
+        # Christmas Day.
+        name = tr("Hari Natal")
+        dt_christmas = date(self._year, DEC, 25)
+        self._add_observed(dt_christmas, rule=SAT_SUN_TO_NEXT_MON)
+        self._add_christmas_day(name)
+
+        # Boxing Day.
+        name = tr("Hari Boxing")
+        dt_boxing = date(self._year, DEC, 26)
+        self._add_christmas_day_two(name)
+        boxing_weekday = dt_boxing.weekday()
+        if boxing_weekday == MON:
+            self._add_observed(dt_boxing, name=name, rule=MON_TO_NEXT_TUE)
+        elif boxing_weekday in (SAT, SUN):
+            self._add_observed(dt_boxing, name=name, rule=SAT_SUN_TO_NEXT_MON_TUE)
```

Comment:
```suggestion
        self._add_observed(
            # Boxing Day.
            self._add_christmas_day_two(tr("Hari Boxing")),
            rule=SAT_SUN_TO_NEXT_MON_TUE + MON_TO_NEXT_TUE,
        )
```

Reply:
The holidays also need to be ordered: first, according to Gregorian calendar (chronologically), then Islamic holidays (chronologically by Hijri).

Reply:
❗